subsys/bluetooth/host/monitor.c

```diff
@@ -328,6 +328,8 @@ static int bt_monitor_init(struct device *d)
 
 	monitor_dev = device_get_binding(CONFIG_BT_MONITOR_ON_DEV_NAME);
 
+	__ASSERT_NO_MSG(monitor_dev);
```

Comment:
If asserts are not enabled we would give monitor_dev to uart_irq_rx/tx_disable below. Maybe we should do the handling that you did in shell_uart.c in all places that you added the assert?

Reply:
This was my initial intention, but in functions invoked in `SYS_INIT` macro (like current `bt_monitor_init`), returned values won't be checked AFAIU.

Reply:
That is strange, seems wierd to have return codes then. But the return would stop you from doing something harmful at least, it could be a null pointer write which would lead to MPU fault or other types of memory errors. And maybe somebody will check the return value in the future :)

Reply:
> But the return would stop you from doing something harmful at least, it could be a null pointer write which would lead to MPU fault or other types of memory errors. 

I'd rather not do this as this would likely be easily caught by various code analysis scripts, so this would add support follow-up to this PR.

> And maybe somebody will check the return value in the future :)

Or... somebody will require these functions to be void :-)

Reply:
well I'm just skeptical since the ASSERT might not be enabled. And then the check would simply be skipped an the code continues, so we would end up in undefined behavior anyway.

Reply:
I agree. ASSERT are disabled by default, but in case one started to debug this kind of things, enabling them would be one of the first step (was mine at least).
Anyway the real solution to this is solving Kconfig/dt desynchro and @galak is working on it.